curvefs/src/mds/chunkid_allocator.cpp

```diff
@@ -26,20 +26,24 @@
 namespace curvefs {
 namespace mds {
 
-int ChunkIdAllocatorImpl::GenChunkId(uint64_t *chunkId) {
+int ChunkIdAllocatorImpl::GenChunkId(uint64_t idNum, uint64_t *chunkId) {
     int ret = 0;
     // will unloack at destructor
     ::curve::common::WriteLockGuard guard(nextIdRWlock_);
 
-    if (nextId_ > lastId_ || CHUNKIDINITIALIZE == nextId_) {
+    if (nextId_ + idNum > lastId_ || CHUNKIDINITIALIZE == nextId_) {
```

Comment:
nextId_ + idNum - 1 > lastId_ ? 
latestid_ indicates the last assignable id, and netxid indicates the next assignable id. So the current number of assignable ids is latestid - nextid +1.